src/Common/Contracts/Repository_Interface.php

```diff
@@ -0,0 +1,176 @@
+	 * the primary key.
+	 *
+	 * By default the primary key is the post ID.
+	 *
+	 * @param mixed $primary_key The primary key value.
+	 *
+	 * @return Model|null|WP_Post
+	 */
+	public function by_primary_key( $primary_key );
+
+	/**
+	 * Deletes a set of events fetched by using filters.
+	 *
+	 * @since 4.9.5
+	 *
+	 * @param bool $return_promise Whether to return the promise or just the deleted post IDs
+	 *                             if the deletion happens in a background process; defaults
+	 *                             to `false`.
+	 *
+	 * @return int[]|Promise An array of deleted post IDs, or that will be deleted in asynchronous
+	 *                       mode or a promise object if `$return_promise` is set to `true`. The
+	 *                       promise object will immediately execute its resolved or rejected callback
+	 *                       if in synchronous mode.
+	 */
+	public function delete( bool $return_promise = false );
+
+	/**
+	 * Commits the updates to the selected post IDs to the database.
+	 *
+	 * @since 4.7.19
+	 *
+	 * @param bool $return_promise Whether to return a promise object or just the ids
+	 *                             of the updated posts; if `true` then a promise will
+	 *                             be returned whether the update is happening in background
+	 *                             or not.
+	 *
+	 * @return array|Promise A list of the post IDs that have been (synchronous) or will
+	 *                       be (asynchronous) updated if `$return_promise` is set to `false`;
+	 *                       the Promise object if `$return_promise` is set to `true`.
+	 */
+	public function save( $return_promise = false );
+
+	/**
+	 * Adds an alias for an update/save field.
+	 *
+	 * @since 4.9.5
+	 *
+	 * @param string $alias The alias to add.
+	 * @param string $field_name The field name this alias should resolve to, this
+	 *                           can be posts table field, a taxonomy name or a custom
+	 *                           field.
+	 */
+	public function add_update_field_alias( $alias, $field_name );
+
+	/**
+	 * Returns the update fields aliases for the repository.
+	 *
+	 * @since 4.9.5
+	 *
+	 * @return array This repository update fields aliases map.
+	 */
+	public function get_update_fields_aliases();
+
+	/**
+	 * Replaces the update fields aliases for this repository.
+	 *
+	 * @since 4.9.5
+	 *
+	 * @param array $update_fields_aliases The new update fields aliases
+	 *                                     map for this repository.
+	 */
+	public function set_update_fields_aliases( array $update_fields_aliases );
+
+	/**
+	 * Creates a post of the type managed by the repository with the fields
+	 * provided using the `set` or `set_args` methods.
+	 *
+	 * @since 4.9.5
+	 * @since TBD Added Model as a possible return type.
+	 *
+	 * @return WP_Post|Model|false|null The created post object or `false` if the creation
```

Comment:
The comment mentions objects or `false`, but what about returning `null`?

Do we need an `@throws` tag?

Reply:
I think the null was just a mistake. i couldn't see it so i removed it. I added the throws tag.